### ISSUE TITLE
GGMAX ENHANCEMENT - parenting/constraints system for the importer

### ISSUE THREAD

**Issue by mav3r1ck1981** [TheGameCreators/GameGuruRepo#4993]
I get that this is an enhancement and potentially low down on the list. However, i do believe that it will be important as max progresses and games begin to be released.

As it is now, custom characters are available to be created using the latest max rig provided by Bond1, and custom melee weapons can be imported. Both work well together and weapons work well with for the player to utilise. BUT neither process is particularly easy, or quick and if we are honest, potentially out of the abilities of casual or newer users. This breeds frustrations, and means that a large proportion of your userbase are beholden to third party artists creating DLCs. You will notice that at present there is a distinct lack of weapon AND NPC/character creator parts on the market, and this is in no small part due to the laborious process of creating them.

A great alternative is Necryms NPC_controller. It allows for good functionality for enemy/ally and neutral NPCs and can be used not only for combatants, but for vendors/roving characters/mounts to name but a few.

An animated character that has been purchased (for example) often comes with weapons/torches etc, basically items that will be held in the hand, and more often than not, the items will be animated to work with the character using constraints or parenting. At present, Max appears to "half" recognise these constraints, as shown in the video below:

https://github.com/TheGameCreators/GameGuruRepo/assets/61555547/1ccc4253-4674-4ead-a708-df141a1ec16a

Max is soooooooooooooooooo close with this. If there was a was that mac could fully recognise the constaints binding an object to a character, or better yet, a built in system in the importer that would allow for the connecting of items to a character if would allow users to dramatically customise their games and more importantly the end-user experience. Perhaps most importantly, it would provide your userbase with a real sense of achievement, allowing us to progress will REAL tangible effects on the worlds we create. Its not only about combat, its useage is wide and varied: think being hunted underground in caverns with orcs holding torches searching for you, think blacksmiths forging weapons with hammers and tong, scifi NPCs typing on virtual tablets over even spiders climbing from the roof on a thread of silk to attack the player. The combat scenarios are obvious, but the ability to "add" held items to an NPC without having to learn 3d modelling/animation/weapon importing/LUA etc has the potential to open up a vast world of customising to ALL of your users, not just the highly skilled.

For me at least, this goes hand in hand with "dream it, build it, play it" and for TGC and max, it means excited, engaged and an active community of game makers extolling the virtues of Max. We all want it to be great and its so close. 

Thanks for listening to my TED talk.....i hope ive explained myself well

**Comment by smallg2020** [comment]
yh this would be cool, give us the ability to attach to any existing bone please
major bonus points if it gets opened up to lua so items can be swapped in and out during the game (i.e. NPC can swap weapon visual or upgrade their armour etc).

**Comment by mav3r1ck1981** [comment]
Absolutley this @smallg2020

**Comment by MonkeyFrogStudio** [comment]
We definitely need a way to more easily use characters as NPCs in MAX ... not only those that can be downloaded or purchased, but those we create ourselves. Character models for games often use weapons and these weapons are their own models, animated separately to work with the NPC ... often to work with a variety of NPCs. As a result, we need a method to ensure these weapons work with these NPCs.

**Comment by C4ever08** [comment]
I agree with Maverick. Its not an easy task and can be a bit frustrating. And to have more options to get in are own NPC custom characters in Max would open up more creative ideas. I believe a parenting/constraints system is a good idea.

**Comment by GraPhiX-Guru** [comment]
This gets my vote, i think a being able to attach to bones would be great especially if you can import a third party/custom rig and match it to MAX rig so that it could be used, with the option of weapon attachment bone in both hands
